packages/language-support/src/highlighting/syntaxValidation/syntaxValidation.ts

```diff
@@ -163,57 +161,62 @@ export function sortByPosition(a: SyntaxDiagnostic, b: SyntaxDiagnostic) {
 }
 
 export function lintCypherQuery(
-  wholeFileText: string,
+  query: string,
   dbSchema: DbSchema,
 ): SyntaxDiagnostic[] {
-  const syntaxErrors = validateSyntax(wholeFileText, dbSchema);
-
+  const syntaxErrors = validateSyntax(query, dbSchema);
   if (syntaxErrors.length > 0) {
     return syntaxErrors;
   }
 
-  return validateSemantics(wholeFileText);
+  const semanticErrors = validateSemantics(query);
+  return semanticErrors;
 }
 
 export function validateSyntax(
   query: string,
   dbSchema: DbSchema,
 ): SyntaxDiagnostic[] {
-  if (query.length > 0) {
-    const parsingResult = parserWrapper.parse(query);
-    const diagnostics = parsingResult.diagnostics;
-
-    const labelWarnings = warnOnUndeclaredLabels(parsingResult, dbSchema);
-    return diagnostics.concat(labelWarnings).sort(sortByPosition);
+  if (query.length === 0) {
+    return [];
   }
+  const statements = parserWrapper.parse(query);
+  const result = statements.statementsParsing.flatMap((statement) => {
+    const diagnostics = statement.diagnostics;
+    const labelWarnings = warnOnUndeclaredLabels(statement, dbSchema);
+    return diagnostics.concat(labelWarnings).sort(sortByPosition);
+  });
 
-  return [];
+  return result;
 }
 
-export function validateSemantics(wholeFileText: string): SyntaxDiagnostic[] {
-  const reparse = parserWrapper.parse(wholeFileText);
-  if (reparse.diagnostics.length > 0) {
-    return [];
-  }
-
-  /*  
-    Semantic analysis can only handle one cypher statement at a time and naturally only supports cypher.
-    We work around these limitations by breaking the file into statements, then run semantic analysis 
-    on each individual cypher statement and map the positions back to the original query.
-    */
-  return reparse.collectedCommands
-    .flatMap((cmd) => {
-      if (cmd.type === 'cypher' && cmd.query.length > 0) {
-        const { notifications, errors } = wrappedSemanticAnalysis(cmd.query);
-
-        return fixSemanticAnalysisPositions({
-          cmd,
-          semanticElements: notifications.concat(errors),
-          parseResult: reparse,
-        });
+/**
+ * Assumes the provided query has no parse errors
+ */
```

Comment:
Since we do the parse here anyway, why assume that there are no parse errors when we can just check `cachedParse.diagnostics`?